### DISCUSSION TITLE
Does Sharppcap work with AOH compilation?

### DISCUSSION THREAD

**Discussion by gardc** [chmorgan/sharppcap#481] [Q&A] [answered]
I've been thinking of using Sharppcap with C# as an alternative to PcapPlusPlus for making a cross platform sniffing tool, but I don't want the user to install .NET runtime in order to use the tool. That's why I'd like it to AOT compile. Does Sharppcap support this?

**Comment by kayoub5** [comment]
Windows or Linux?

**Reply by gardc** [reply]
both, + macos

**Comment by gardc** [comment]
After reading https://learn.microsoft.com/en-us/dotnet/core/deploying/ , let me rephrase: is it possible to use sharppcap in AOT compile mode OR self containing mode, meaning the .NET runtime is shipped with the produced binary?

**Reply by kayoub5** [reply, marked as answer]
currently it works in Windows, but not in Linux (due to different library name in each platform), there is PR  #448, that makes it work in Linux, but it breaks Windows support.

**Reply by gardc** [reply]
alright thanks

**Comment by kayoub5** [comment]
please keep in mind that if you use self-contained app, the LGPL license of sharppcap, requires that the resulting binary is also distributed under LGPL.

**Reply by gardc** [reply]
oh ok, I'll keep that in mind

**Comment by chmorgan** [comment]
Didn’t we talk about relicensing at one point? Thoughts on the topic Ayoub?

**Reply by kayoub5** [reply]
yes, but that was for packetnet, not sharppcap, it would take a bit of effort to get every major contributor email and contact them about changing the license, but I don't expect any opposition.

**Reply by gardc** [reply]
For my sake, as one potential user of this (awesome) lib bear in mind, that would make things a lot easier if it was MIT licensed for instance
